cli/internal/helm/client.go

```diff
@@ -75,7 +75,7 @@ func NewClient(client crdClient, kubeConfigPath, helmNamespace string, log debug
 	return &Client{kubectl: client, fs: fileHandler, actions: actions{config: actionConfig}, log: log}, nil
 }
 
-func (c *Client) shouldUpgrade(releaseName, newVersion string) error {
+func (c *Client) shouldUpgrade(releaseName, newVersion string, force bool) error {
```

Comment:
The helm chart logic is still not doing what I would expect it to do:

What I want to happen:

if currentVersion == newVersion => already at target version, do nothing (do not perform an upgrade)
if (currentVersion != newVersion) && force => perform upgrade
if (currentVersion != newVersion) && isValidUpgrade(currentVersion, newVersion) => perform upgrade
else: not compatible (do not perform an upgrade)

What this PR would end up doing:

if force: perform upgrade
if !force && isValidUpgrade(currentVersion, newVersion) => perform upgrade
else: not compatible (do not perform an upgrade)